src/main/java/bio/terra/service/filedata/FileService.java

```diff
@@ -177,6 +183,71 @@ public String ingestBulkFileArray(
         .submit();
   }
 
+  public List<FileModel> listDatasetFiles(String datasetId, Integer offset, Integer limit) {
+    Dataset dataset = datasetService.retrieve(UUID.fromString(datasetId));
+    CloudPlatformWrapper cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    String collectionId = datasetId;
+    List<FireStoreFile> results;
+    if (cloudPlatformWrapper.isGcp()) {
+      try {
+        collectionId = String.format("%s-files", datasetId);
+        results = fileDao.retrieveFiles(dataset, collectionId, offset, limit);
+      } catch (InterruptedException | ExecutionException ex) {
+        throw new FileSystemExecutionException(
+            "Unexpected interruption during file system processing", ex);
+      }
+    } else {
+      collectionId = FILES_TABLE.toTableName(UUID.fromString(collectionId));
+      BillingProfileModel billingProfileModel =
+          profileService.getProfileByIdNoCheck(dataset.getDefaultProfileId());
+      AzureStorageAccountResource storageAccountResource =
+          resourceService.getDatasetStorageAccount(dataset, billingProfileModel);
+      AzureStorageAuthInfo storageAuthInfo =
+          AzureStorageAuthInfo.azureStorageAuthInfoBuilder(
+              billingProfileModel, storageAccountResource);
+      results = tableDao.listFiles(collectionId, storageAuthInfo, offset, limit);
+    }
+    String finalCollectionId = collectionId;
+    return results.stream()
+        .map(f -> FireStoreFile.toFileModel(f, finalCollectionId))
+        .collect(Collectors.toList());
```

Comment:
Can use `toList()` here
```suggestion
        .toList();
```